OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/ModelStore.kt

```diff
@@ -156,25 +158,54 @@ abstract class ModelStore<TModel>(
         changeSubscription.fire { it.onModelRemoved(model, tag) }
     }
 
+    /**
+     * When models are loaded from the cache, they are added to the front of existing models.
+     * This is primarily to address operations which can enqueue before this method is called.
+     */
     protected fun load() {
         if (name == null || _prefs == null) {
             return
         }
 
+        val shouldRePersist = models.isNotEmpty()
```

Comment:
This should be in the `synchronized(models) {}` below to prevent race conditions and concurrent modification expectations.